includes/Checker/Abstract_Check_Runner.php

```diff
@@ -287,6 +325,11 @@ final public function get_checks_to_run() {
 			$check_flags = Check_Repository::TYPE_ALL;
 		}
 
+		// Check whether to include experimental checks.
+		if ( $this->include_experimental ) {
```

Comment:
This doesn't work reliably: For the same reason we call the `get_check_slugs()` method above instead of accessing `$this->check_slugs`, we have to call a method that ensures the parameter is retrieved instead of directly accessing `$this->include_experimental`.

We need to introduce a `get_include_experimental()` method which works equivalent to the `get_check_slugs()` method: If the class property isn't `null`, return that, otherwise return the result of `get_include_experimental_param()`.

Reply:
Thanks @felixarntz I've added the new `get_include_experimental` method